src/new/swapConnection.ts

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import BaseConnection, {
+  ConnectionParams,
+} from '../connection';
+import {
+  ConnectionState,
+} from '../wallet/types';
+import AtomicSwapWebSocket from '../websocket/atomic-swap';
+
+
+/**
+ * This is a Websocket Connection with the Atomic Swap Service
+ *
+ * It has the following states:
+ * - CLOSED: When it is disconnected from the server.
+ * - CONNECTING: When it is connecting to the server.
+ * - CONNECTED: When it is connected.
+ *
+ * You can subscribe for the following events:
+ * - update-atomic-swap-proposal: Fired when the state of a listened proposal changes
+ **/
+export class AtomicSwapServiceConnection extends BaseConnection {
```

Comment:
Why are we extending `BaseConnection` when `BaseConnection` was meant to be a ws connection with the fullnode?

Reply:
Good point! Opened #535 to discuss the refactoring possibilities for this, including the Wallet Service websocket that is already implemented using it.

Reply:
I agree with @r4mmer that it's a different use. The `BaseConnection` is not meant to be used with full node only but with wallet context.

I think it's bad to use this as base class, you should refactor it before merging, it's weird that you have your `ConnectionParams` options with `network` and `servers`, which makes no sense for the swap service.